src/Polygon2d.h

```diff
@@ -41,3 +41,18 @@ class Polygon2d : public Geometry
 	Outlines2d theoutlines;
 	bool sanitized;
 };
+
+class Polygon2dIn3d : public Polygon2d
+{
+public:
+	Polygon2dIn3d(const Polygon2dIn3d &p): Polygon2d(p), matrix(p.matrix) {}
+	Polygon2dIn3d(const Polygon2d &p, const Transform3d &m): Polygon2d(p), matrix(m) {}
+	Polygon2dIn3d(const Polygon2d &p): Polygon2d(p), matrix(Transform3d::Identity()) {}
+
+	size_t memsize() const override;
+	std::string dump() const override;
+	Geometry *copy() const override { return new Polygon2d(*this); }
+	unsigned int getDimension() const override { return 23; }
```

Comment:
This is quite the hack. If we decide to go down this route, we should rather change dimension to be a type-safe enum.

Reply:
I agree :-)  actually my latest idea is to make a patch that does "lazy 2D transforms" and then combined with your patch that does "lazy unions", this extrusion patch would be a lot simpler.  Also I accidentally just force-pushed this branch and didn't mean to.

Reply:
Force-pushing to your own feature branches is not a problem; I tend to do the same to be able to clean up commits.